File: ConvexOptimization/long_only_ml_data.m
n = 20;
rng(0)
%Sigma is the true covariance - PSD
Sigma = rand(n, n);
Sigma = 0.5*(Sigma + Sigma');
Sigma = Sigma + n*eye(n);
% Res = (Sigma^{-1} temp_mu)_+
% Designed so that there are some zero entries
temp_mu = rand(n,1);
res = Sigma \ temp_mu;
res = max(res, 0);
% Compute implied mu
mu = Sigma*res;
% Draw N random samples from distribution
N = 25;
X = randn(n, N);
MU = zeros(n, N);
for i = 1:N
MU(:,i) = mu;
end
Sq = sqrtm(Sigma);
X = Sq*X;
X = X + MU;

File: ConvexOptimization/long_only_ml_data.py
np.random.seed(364)
N = 25
n = 2
# Sigma is the true covariance
Sigma = np.random.uniform(size=(n, n))
Sigma = (1 / 2) * (Sigma + Sigma.T)
Sigma = Sigma + n * np.eye(n)
# mu is the true mean
temp_mu = np.random.uniform(size=(n, 1))
res = np.linalg.inv(Sigma) @ temp_mu
res = np.maximum(res, 0)
mu = (Sigma @ res).reshape((n,))
323
# X is drawn from Normal(mu, Sigma_
X = np.random.multivariate_normal(mean=mu, cov=Sigma, size=N)
